test_simple_graph.py

```diff
@@ -41,9 +69,22 @@ def test_add_edge(populated):
     assert [7, 5] in populated.edges()
 
 
+def test_add_edge_new_node(populated):
+    populated.add_edge(5, 28)
+    assert 28 in populated.graph
+    assert 28 in populated.graph[5]
+
+
 def test_del_node(populated):
     populated.del_node(5)
     assert 5 not in populated.graph
+    for k, v in populated.graph.iteritems():
+            assert 5 not in v
+
+
+def test_del_nonexistent_node(populated):
+    with pytest.raises(KeyError):
+        populated.del_node(70)
```

Comment:
Test that your KeyError returns your custom error message - that will ensure that you are hitting YOUR error and not a different KeyError. Also, I think this test might pass even if there isn't an error, so this test doesn't tell you anything.